standards/application/relay-reliability.md

```diff
@@ -0,0 +1,150 @@
+---
+title: RELIABILITY-RELAY
+name: Reliability for Relay Protocol
+category: Standards Track
+tags: [reliability, application]
+editor: Kaichao Sun <kaichao@status.im>
+contributors:
+  - Richard Ramos <richard@status.im>
+---
+
+## Abstract
+
+[Relay Protocol](https://github.com/vacp2p/rfc-index/blob/main/waku/standards/core/11/relay.md) in Waku is efficient for routing messages, but there's no guarantee that a message will reach to its destination. For example, the receiver in a chat application may miss messages when network issue happens at either sender or receiver side. In general, a message in Waku network includes 3 status:
+
+- **outgoing**, the message is posted by its creator but no confirmations from other nodes yet
+- **sent**, the message is received by any other node in the network
+- **delivered**, the message is acknowledged by the receiver
+
+Application like Status already uses [MVDS](https://github.com/vacp2p/rfc-index/blob/main/vac/2/mvds.md) for e2e acknowledgement in direct messages and group chat. There is an ongoing [discussion](https://forum.vac.dev/t/end-to-end-reliability-for-scalable-distributed-logs/293) about a more general and bandwidth efficient solution for e2e reliablity.
+
+Before we have a complete design for e2e reliability, we need to compose existing protocols to increase the reliability of the relay protocol. This document proposes a few options for such composition.
+
+## Motivation
+
+The [Store protocol](https://github.com/waku-org/specs/blob/master/standards/core/store.md) provides a way for nodes in the network to query the existence of messages or fetch specific messages based on the search criteria.
+
+**Search criteria with message hash**
+
+For the nodes that may have connection issues to **publish** messages via relay network, this search criteria can be used to check whether a message is populated in the network or not. The message exists in store node can be marked from `outgoing` to `sent` by application. If the message is not found in the store node, the application can resend the message.
+
+**Search criteria with topics and time range**
+
+For the nodes that may have connection issues to **receive** messages via relay network, this search criteria can be used to fetch missing messages from store nodes when network resumes. 
```

Comment:
"when network resumes"

As far as I understand the implementation further down, this is not triggered only when a connection resumes but periodically (in order to also cover undetected "offline" periods)?

Reply:
Yes, this is executed periodically regardless of the network connectivity state